### PR TITLE
Implement more math commands (e.g. trig, ln)

### PR DESCRIPTION
**Pull request by sholderbach**
# Description

Work towards #7073	

- Add `math pi` and `math e` constants
- Add basic trigonometric functions
  - `math sin`
  - `math cos`
  - `math tan`
- Add basic hyperbolic functions
  - `math sinh`
  - `math cosh`
  - `math tanh`
- Add inverse for trigonometric functions  
    - `math arcsin`
    - `math arccos`
    - `math arctan`
- Add inverse hyperbolic functions    
    - `math arcsinh`
    - `math arccosh`
    - `math arctanh`
- Add natural logarithm
    - `math ln`  

# Tests + Formatting

- Examples for each new command
- Expand example test harness with `math pi`, `math e`, and `math round`

### CODE REVIEW

File: crates/nu-command/src/default_context.rs [hunk 1]
```diff
@@ -405,6 +405,21 @@ pub fn create_default_context() -> EngineState {
             MathStddev,
             MathSum,
             MathVariance,
+            Sin,
```
Comment (fdncred) [inline crates/nu-command/src/default_context.rs:408]: The only thing that jumped out at me was tha tall these internal names start with Math and then these new ones do not start with Math. I wasn't sure if that was by design or just an oversight?

**Comment by fdncred** [comment]
Lots of repetitive code here, as is the way when adding a bunch of commands. I skimmed through these, mainly looking for copy-n-paste errors and such, and none jumped out at me. I think we should move ahead with landing these and your other math pr too, and then take a look at combining all this stuff in a math plugin. I think it's the only way we'll be able to do any meaningful perf testing to determine if moving to a plugin is the right thing to do.

File: crates/nu-command/src/default_context.rs (hunk 1 above)
Comment (sholderbach) [reply crates/nu-command/src/default_context.rs:408]: Yeah just a bit of rebellion to the repetitiveness :D I can make that consistent.

**Comment by sholderbach** [comment]
Thanks for going through all the boilerplate @fdncred!

Should I rebase the blocks of commands into their own commits so that removing them is easier if we decide to move them to a plugin?

**Comment by fdncred** [comment]
> Should I rebase the blocks of commands into their own commits so that removing them is easier if we decide to move them to a plugin?

That sounds like a pain. I probably wouldn't.

**Comment by sholderbach** [comment]
> > Should I rebase the blocks of commands into their own commits so that removing them is easier if we decide to move them to a plugin?
> 
> That sounds like a pain. I probably wouldn't.

I was a masochist in this case.

**Pull request merged**